### ISSUE TITLE
AI Origin & Provenance Lab: local multi-stream AI-music source detector

### ISSUE THREAD

**Issue by rassvetpublic-spec** [rassvetpublic-spec/Genre_test#80]
## Goal
Build a local-first, evidence-traceable **AI Origin Super Detector** for `Genre_test` that runs reliably on the verified target workstation (RTX 5070 Ti 16 GB, Blackwell `sm_120`, PyTorch 2.12.1+cu130 / CUDA 13.0).

This is an independent provenance/origin subsystem. It must not change `AudioProfile`, `GenerativeDefectProfile`, mastering quality, or existing analyzer `DRIFT` semantics.

## Product output
Introduce versioned `OriginProfileV1` with explicit uncertainty:

- `HUMAN_LIKELY`
- `AI_LIKELY`
- `MIXED_OR_PARTIAL`
- `UNCERTAIN`
- `INSUFFICIENT_EVIDENCE`

Source-family attribution is hierarchical and only published when supported:

- `HUMAN`
- `SUNO`
- `UDIO`
- `MUSICGEN`
- `STABLE_AUDIO`
- `AUDIOLDM`
- `RIFFUSION`
- `OTHER_AI`
- `UNKNOWN`

Absence of known AI fingerprints is **not** proof of human origin.

## Target multi-stream architecture

```text
ORIGINAL AUDIO
  -> native forensic path
     -> Fourier fakeprint / decoder artifacts
     -> CQT forensic CNN
     -> optional codec/residual/phase features
  -> semantic/structure path
     -> existing pinned MERT 95M segment embeddings
     -> verified-human novelty / OOD stream
     -> long-context sequence detector
     -> hierarchical source attribution
  -> optional provenance path
     -> metadata / supported provenance markers
  -> calibrated evidence fusion
  -> conformal uncertainty / FPR-controlled verdict
  -> OriginProfileV1
```

## Critical implementation rules

1. **Reuse the proven MERT runtime.** Do not introduce a second heavyweight MERT stack unless benchmark evidence proves it necessary. Current #29 runtime is already verified on RTX 5070 Ti.
2. **Native/high-rate forensic path.** Do not use only the 16 kHz MAEST stream for spectral forensic evidence.
3. **One decode / multiple views** where practical: native/44.1–48 kHz forensic, 24 kHz MERT, 16 kHz core analysis.
4. Heavy stages execute sequentially under the future `ModelRuntimeManager` / VRAM ownership contract; do not keep every model resident simultaneously.
5. AI-origin evidence is never used as a mastering/repair optimization target and the project does not implement detector evasion, watermark removal, or provenance concealment.

### Long-context overlap
Default long-context segmentation target:

```text
window = 5.0 s
stride = 2.5 s
```

The 50% overlap is mandatory in the benchmark because non-overlapping windows can miss synthetic splice/transition boundaries. Segment policy must remain configurable and benchmarked against 5 s / 5 s non-overlap.

### CQT codec robustness
Training/robustness augmentation must include variable streaming-codec conditions rather than a single MP3 setting:

- MP3 multiple bitrates;
- AAC multiple bitrates/profiles where reproducible;
- Opus multiple bitrates;
- resampling 44.1/48/24/16 kHz paths as appropriate;
- ordinary EQ/compression/limiting/loudness normalization;
- phase shifts / phase rotation / channel-phase perturbations that preserve ordinary listening use-cases;
- controlled noise where useful.

The benchmark must measure both masking of true AI evidence and codec-induced false positives on verified-human controls.

### Verified Human purity
The verified-human reference set must not be treated as one homogeneous distribution. Old digitizations, remasters, lossy archives and unusual production chains can become false OOD outliers.

Required corpus strata include, where data permits:

- native/lossless modern human masters;
- modern remasters;
- legacy masters / older digitizations;
- lossy-source human tracks;
- acoustic/live material;
- electronic/heavily processed human material;
- genre-balanced and artist-disjoint calibration/test subsets.

`verified_human` requires provenance evidence; unknown catalog tracks must not silently enter one-class training/calibration.

Novelty calibration must test per-stratum and per-genre FPR before promotion.

## Benchmark contract

Primary operating metric:

- **Recall @ 1% verified-human FPR**

Additional required metrics:

- Recall @ 0.1% FPR
- Recall @ 5% FPR
- ROC-AUC / PR-AUC
- Brier score / calibration error
- per-generator and worst-generator cells
- per-genre and per-human-stratum FPR
- score/verdict drift under codec/mastering transformations

Primary generalization protocol:

- **LOGO — Leave One Generator Out**
- artist/song-disjoint human partitions
- external holdout sets not used for training
- content-matched pairs where possible to reduce corpus/mastering confounds

Average performance must never hide the worst held-out generator.

## Runtime gates for RTX 5070 Ti 16 GB

Target production inference budget:

- normal peak VRAM target: `< 8 GB`
- hard gate before promotion: `< 11 GB`
- preserve safety headroom for the wider SUPERCOMBINE runtime
- no hidden model downloads
- clean release of GPU state after optional heavy stages
- structured OOM/fallback errors

## Initial work packages

- OriginProfile/schema + benchmark truth semantics
- native forensic decode/view foundation
- Deezer-style Fourier fakeprint baseline
- LOGO/FPR benchmark harness
- CQT forensic CNN + codec/phase robustness
- shared MERT segment cache with 5 s / 2.5 s overlap experiments
- long-context sequence detector
- verified-human novelty/OOD detector
- hierarchical generator attribution
- calibrated multi-stream fusion + conformal uncertainty
- GUI/CLI/history/runtime integration after detector gates pass

## Promotion rule
A new stream joins production fusion only if it improves the locked benchmark, especially **worst-generator Recall@1%FPR**, or materially reduces human false positives/uncertainty without unacceptable regression.

## Merge rule
No merge without explicit **MTD**.

**Comment by rassvetpublic-spec** [comment]
Accepted AI Origin & Provenance Lab plan is now fixed in documentation PR #89.

Issue map:
- #81 OriginProfileV1 + LOGO/fixed-FPR benchmark
- #82 native forensic views + Fourier fakeprint baseline
- #83 CQT CNN with variable MP3/AAC/Opus bitrate + phase robustness
- #84 shared MERT segment cache + 5.0 s window / 2.5 s stride long-context detector
- #85 Verified Human novelty/OOD with remaster/legacy-digitization/lossy/live/electronic strata and per-stratum FPR
- #86 hierarchical generator attribution with OTHER_AI/UNKNOWN
- #87 calibrated multi-stream fusion + conformal uncertainty
- #88 local runtime/CLI/GUI/history integration and RTX 5070 Ti gates

Docs in PR #89:
- `docs/AI_ORIGIN_PROVENANCE_LAB.md`
- `docs/AI_ORIGIN_ROADMAP.md`

Runtime target remains local RTX 5070 Ti 16 GB, normal origin peak `< 8 GB`, hard promotion gate `< 11 GB`, reusing the already verified MERT 95M runtime from #29.

No merge without explicit MTD.

**Comment by rassvetpublic-spec** [comment]
## Architecture amendment — Detector Evidence Lab #198

Detector robustness research is explicitly **in scope** for Genre_test. #198 now owns the evidence/disclosure layer around this detector architecture.

A detector failure case may be retained as a first-class `DetectorEvidenceSample`: immutable source hash, evidence-WAV hash, known ground truth, detector/version/date, raw score/verdict, complete processing history, runtime/tool identities and reproduction status. A WAV of known origin that a detector fails to identify can therefore be valid project evidence when the experiment is fully documented.

Keep this Issue's calibrated origin-model semantics. Do not treat any detector as provenance ground truth and do not let ordinary repair/mastering silently optimize against a detector score.

Evidence-audio rights are separate from repository code licensing: #198 requires default prohibition on commercial exploitation of evidence WAVs as music/media while explicitly allowing detector developers to use them for internal validation, remediation, responsible disclosure and bounty/research-reward review.
